src/RSocketRequester.cpp

```diff
@@ -35,6 +35,7 @@ RSocketRequester::RSocketRequester(
 
 RSocketRequester::~RSocketRequester() {
   LOG(INFO) << "RSocketRequester => destroy";
+  stateMachine_->close(folly::exception_wrapper(), StreamCompletionSignal::CONNECTION_END);
```

Comment:
stateMachine_'s destructor has a check 
  DCHECK(isDisconnectedOrClosed()); // the instance should be closed by via

Do you observe if this check is failing if you do not call the close() method in here? 
Should we call the close() method at the destructor of the stateMachine_ ?

I saw that in the close() method of the RSocketStateMachine, we call lots of callback functions. Any of these functions might fail and if might cause early termination of the function. But at the start of the function we call
  isClosed_ = true;
Should we place some try/catch in the close function and place the isClosed_ = true at the end of this function?